### ISSUE TITLE
URLTextSearcher: Error: 'utf-8' codec can't decode byte 0x85 in position 61966: invalid start byte

### ISSUE THREAD

**Issue by macmule** [autopkg/autopkg#619]
**Describe the problem**
URLTextSearcher under AutoPkg 2.x is erroring out under some recipes with:

```Error: 'utf-8' codec can't decode byte 0x85 in position 61966: invalid start byte```

So far seen this with ImageJ & VIA recipes.

**Preferences contents**

```
{
    "GIT_PATH" = "/usr/local/git/bin/git";
    "MUNKI_REPO" = "/Users/Shared/munki_repo";
    "RECIPE_REPOS" =     {
        "/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes" =         {
            URL = "https://github.com/autopkg/dataJAR-recipes.git";
        };
        "/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes" =         {
            URL = "https://github.com/autopkg/homebysix-recipes.git";
        };
        "/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes" =         {
            URL = "https://github.com/autopkg/recipes.git";
        };
    };
    "RECIPE_SEARCH_DIRS" =     (
        ".",
        "~/Library/AutoPkg/Recipes",
        "/Library/AutoPkg/Recipes",
        "/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes",
        "/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes",
        "/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes"
    );
}
```

**AutoPkg output**

```
autopkg run -vvv ~/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes/VIA/VIA.download.recipe 
Processing /Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes/VIA/VIA.download.recipe...
WARNING: /Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes/VIA/VIA.download.recipe is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
{'AUTOPKG_VERSION': '2.0.2',
 'GIT_PATH': '/usr/local/git/bin/git',
 'MUNKI_REPO': '/Users/Shared/munki_repo',
 'NAME': 'VIA',
 'PARENT_RECIPES': [],
 'RECIPE_CACHE_DIR': '/Users/ben/Library/AutoPkg/Cache/com.github.dataJAR-recipes.download.VIA',
 'RECIPE_DIR': '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes/VIA',
 'RECIPE_OVERRIDE_DIRS': ['~/Library/AutoPkg/RecipeOverrides'],
 'RECIPE_PATH': '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes/VIA/VIA.download.recipe',
 'RECIPE_REPOS': {'/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes': {'URL': 'https://github.com/autopkg/dataJAR-recipes.git'},
                  '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes': {'URL': 'https://github.com/autopkg/homebysix-recipes.git'},
                  '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes': {'URL': 'https://github.com/autopkg/recipes.git'}},
 'RECIPE_SEARCH_DIRS': ['.',
                        '~/Library/AutoPkg/Recipes',
                        '/Library/AutoPkg/Recipes',
                        '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes',
                        '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes',
                        '/Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes'],
 'URL': 'https://k.kramerav.com/support/download.asp?f=56977',
 'verbose': 3}
URLTextSearcher
{'Input': {'re_pattern': '(/downloads/setups/via_mac_dmg_v.2.[\\S]+\\.zip)',
           'result_output_var_name': 'match',
           'url': 'https://k.kramerav.com/support/download.asp?f=56977'}}
Traceback (most recent call last):
  File "/Library/AutoPkg/autopkglib/__init__.py", line 673, in process
    self.env = processor.process()
  File "/Library/AutoPkg/autopkglib/__init__.py", line 480, in process
    self.main()
  File "/Library/AutoPkg/autopkglib/URLTextSearcher.py", line 121, in main
    content = self.download_with_curl(curl_cmd)
  File "/Library/AutoPkg/autopkglib/URLGetter.py", line 197, in download_with_curl
    proc_stdout, proc_stderr, retcode = self.execute_curl(curl_cmd, text)
  File "/Library/AutoPkg/autopkglib/URLGetter.py", line 189, in execute_curl
    text=text,
  File "/Library/AutoPkg/Python3/Python.framework/Versions/Current/lib/python3.7/subprocess.py", line 490, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/Library/AutoPkg/Python3/Python.framework/Versions/Current/lib/python3.7/subprocess.py", line 964, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/Library/AutoPkg/Python3/Python.framework/Versions/Current/lib/python3.7/subprocess.py", line 1755, in _communicate
    self.stdout.errors)
  File "/Library/AutoPkg/Python3/Python.framework/Versions/Current/lib/python3.7/subprocess.py", line 841, in _translate_newlines
    data = data.decode(encoding, errors)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x85 in position 61966: invalid start byte
  File "/Library/AutoPkg/autopkglib/__init__.py", line 673, in process
    self.env = processor.process()
'utf-8' codec can't decode byte 0x85 in position 61966: invalid start byte
Failed.
Receipt written to /Users/ben/Library/AutoPkg/Cache/com.github.dataJAR-recipes.download.VIA/receipts/VIA.download-receipt-20200217-135128.plist

The following recipes failed:
    /Users/ben/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes/VIA/VIA.download.recipe
        Error in com.github.dataJAR-recipes.download.VIA: Processor: URLTextSearcher: Error: 'utf-8' codec can't decode byte 0x85 in position 61966: invalid start byte

Nothing downloaded, packaged or imported.
```

**Expected behavior**
Processor should run without errors, as per when ran under AutoPkg 1.x

**Version (please complete the following information):**
 - OS version: 10.15.3
- AutoPkg Version: 2.0.2

**Comment by MichalMMac** [comment]
I can replicate this. Encoding issue. HTML document says it's utf-8 but apparantly it's not.

We could ignore the encoding errors using `errors="ignore"` subprocess option which is passed to [io.TextIOWrapper](https://docs.python.org/3/library/io.html). This fixes this specific issue. Could it have other consequences?

**Comment by macmule** [comment]
Thanks @MichalMMac.

Idk myself what the consequences would be though..

**Comment by macmule** [comment]
I've just made the proposed changes to our URLGetter, thanks @MichalMMac.

Will check back once we've ran out 400 or so recipes this week.

So far, so good.. fixes the issue & already working recipes were fine.

**Comment by macmule** [comment]
400+ recipes ran, no issues.. will PR shortly.

**Comment by macmule** [comment]
@MichalMMac did you want to PR? Or ok with me doing one?

**Comment by MichalMMac** [comment]
@macmule Go ahead. Reference this issue in your PR.

**Comment by macmule** [comment]
Thanks @MichalMMac, done!

**Comment by macmule** [comment]
Issue was closed in PR #629

**Comment by andrewzirkel** [comment]
Looks like this issue is present again as of c8cedfb09f4fdb75bfc19168c715c72c9fd3f30a .  If I add the changes mentioned in this issue the problem is fixed.  I am also using the ImageJ recipe from [jessepeterson](https://github.com/autopkg/jessepeterson-recipes)
